libobs/obs-cocoa.m

```diff
@@ -173,7 +173,9 @@ void log_system_info(void)
 static bool dstr_from_cfstring(struct dstr *str, CFStringRef ref)
 {
     CFIndex length = CFStringGetLength(ref);
-    CFIndex max_size = CFStringGetMaximumSizeForEncoding(length, kCFStringEncodingUTF8);
+    CFIndex max_size = CFStringGetMaximumSizeForEncoding(length, kCFStringEncodingUTF8) + 1;
```

Comment:
I'd like to think that it would more sense to add the 1 byte to `dstr_reserve`, as the `+1` requirement does not stem from the Unicode size requirement, but `dstr`'s API (that still requires the API user to take the terminator size requirement into account.

Reply:
We also need to inform `CFStringGetCString` about the larger buffer size.

If you prefer a separate variable for the required buffer size that's just `max_size + 1` for readability I can do that, just let me know (the compiler would optimize it away anyways so it wouldn't hurt).

Reply:
Right, disregard my comment then, I overlooked `CFStringGetCString`'s requirement to get a buffer size _including_ null terminator.